packages/toolpad-app/src/toolpad/AppEditor/AppEditorShell.tsx

```diff
@@ -111,18 +123,52 @@ export default function AppEditorShell({ appId, ...props }: ToolpadAppShellProps
   const domLoader = useDomLoader();
 
   const [createReleaseDialogOpen, setCreateReleaseDialogOpen] = React.useState(false);
+  const [isSaveStateVisible, setIsSaveStateVisible] = React.useState(false);
+
+  const hasUnsavedChanges = domLoader.unsavedChanges > 0;
+  const isSaving = domLoader.saving;
+
+  React.useEffect(() => {
+    let timeout: NodeJS.Timeout | null = null;
+
+    if (!hasUnsavedChanges) {
+      timeout = setTimeout(() => {
+        setIsSaveStateVisible(false);
+
+        if (timeout) {
+          clearTimeout(timeout);
+        }
+      }, 4500);
+    } else {
+      setIsSaveStateVisible(true);
+    }
+
+    return () => {
+      if (timeout) {
+        clearTimeout(timeout);
+      }
+    };
+  }, [hasUnsavedChanges]);
 
   return (
     <ToolpadAppShell
       appId={appId}
       actions={
         <Stack direction="row" gap={1} alignItems="center">
-          {domLoader.saving ? (
-            <Box display="flex" flexDirection="row" alignItems="center">
-              <CircularProgress size={16} color="inherit" sx={{ mr: 1 }} />
-            </Box>
+          {isSaveStateVisible ? (
```

Comment:
ok, i can use something like `visibility: hidden`

Reply:
What's the reason to hide it exactly?

Reply:
oh i see, i was thinking the save state wasn't that helpful after a while so it could stay hidden - that's also what Bharat proposed. but we can keep the green cloud there as long as it's saved too.